### DISCUSSION TITLE
Creating a data object from JSON string not working

### DISCUSSION THREAD

**Discussion by ferasWb** [spatie/laravel-data#557] [Q&A] [answered]
According to the Documentation [here](https://spatie.be/docs/laravel-data/v3/as-a-data-transfer-object/creating-a-data-object) (Data can also be created from JSON strings: ...), I should be able to create a DataObject from a JSON string without the need for a magic method.

But when I try the exact example from the docs, I'm getting this error:

> Spatie\LaravelData\Exceptions\CannotCreateData  Could not create a `App\Data\SongData` object from value with type `string`, no normalizer was found.

did I misunderstand the docs? or is there a mistake in the docs and I do have to use an explicit magic method for JSON strings?

P.S. I'm using V3.8.1 (the most current at the time)

**Comment by rubenvanassche** [comment]
We have a test testing: this: https://github.com/spatie/laravel-data/blob/7bc0c9564ff7bde825557c3a5713ee754599073e/tests/Normalizers/JsonNormalizerTest.php#L6

Is your JSON valid? Check `JsonNormalizer` for more details.

**Reply by ferasWb** [reply]
yeah, to be sure I used the exact example from the docs.
`
SongData::from('{"name" : "Never Gonna Give You Up","artist" : "Rick Astley"}');
`
and it's causing the same error

**Reply by rubenvanassche** [reply]
Looks like the docs are not correct

> {"name" : "Never Gonna Give You Up","artist" : "Rick Astley"}

should be: {"title" : "Never Gonna Give You Up","artist" : "Rick Astley"}

**Reply by rubenvanassche** [reply]
I've updated it

**Reply by ferasWb** [reply, marked as answer]
Ahh I was missing the JsonNormlizer on my DataObject

```
    public static function normalizers(): array
    {
        return [
            JsonNormalizer::class,
        ];
    }
```

sorry, I'm still new to the lib. And the section of the docs I cited, doesn't mention the need to use the Normalizer when creating from JSON.

Thanks for the quick reply and for pointing me in the right direction!
